main/streams/plain_wrapper.c

```diff
@@ -1114,11 +1120,14 @@ PHPAPI php_stream *_php_stream_fopen(const char *filename, const char *mode, zen
 	}
 
 	if (options & STREAM_ASSUME_REALPATH) {
-		strlcpy(realpath, filename, sizeof(realpath));
+		realpath_len = strlen(filename);
+		ZEND_ASSERT(realpath_len < sizeof(realpath) && "Assumed real path filename is too long");
+		strcpy(realpath, filename);
 	} else {
 		if (expand_filepath(filename, realpath) == NULL) {
 			return NULL;
 		}
+		realpath_len = strlen(realpath);
```

Comment:
This might wasteful - it's not used in all paths below. Just keep `strlen(realpath)` - this function is not really used so not much point to change it.